Sources/Ink/Internal/FormattedText.swift

```diff
@@ -74,6 +77,9 @@ private extension FormattedText {
         }
 
         mutating func parse() {
+            /// For the purpose of hard line break parsing, indicates
+            /// whether the previous two characters were spaces.
+            var twoSpaceSequence = false
```

Comment:
Would an option here be to read ahead instead? That is:
```swift
reader.currentCharacter == " " && reader.nextCharacter == " "
```
If possible, that'd save us this piece of state, which would be nice 🙂

Reply:
The "simple" answer is to read back two characters at line breaks, however that goes against O(*n*). In a sequence (• = space) like `Hello••\n`, we need to understand the `••\n` bit at the `\n`. I chose to read back at the second space since it means only persisting the state one iteration. Otherwise, the state needs to be persisted two loops until the `\n`. **I could instead read back two characters at newlines (checking for •• sequence), which would be simpler but have less linearity.**

Reply:
I've proposed another solution (see it in the code now) that simply tracks sequential spaces, avoiding any backtracking at all. It also means the state is more generic.

Reply:
Very solid reasoning 👍